docs/concepts/client-handshake.mdx

```diff
@@ -0,0 +1,62 @@
+
+With the handshake, we can resolve authentication state on the backend and ensure the request is properly handled as signed in or out, instead of being in a potentially “unknown” state. The handshake flow relies on redirects to exchange session information between FAPI and the application, ensuring the resolution of unknown authentication states minimizes performance impact and behaves consistently across different framework and language implementations.
+
+## Handshake flow
+
+The handshake mechanism operates by way of a redirect from the host application to a FAPI endpoint:
+
+1. Request is made to an application using Clerk
+2. Clerk SDK determines the authentication state of the request (`signed-in`, `signed-out`, or `handshake`).
+3. If authentication state is `handshake`, Clerk responds with a 307 redirect to the handshake endpoint: `fapi/v1/client/handshake`
+4. The handshake endpoint gets information about the current session and returns a handshake payload. The encoded handshake payload contains a list of `Set-Cookie` header directives to be passed along with the final response
+    1. If the session is active, a fresh session JWT cookie is returned
+    2. If the session is inactive, the session JWT cookie is wiped and the request will be treated as signed out
+5. The handshake endpoint redirects back to the host application along with the handshake payload, encoded either in the URL (development) or as a cookie (production)
+6. The handshake payload is parsed and `Set-Cookie` headers are set on the response
+7. If an updated session JWT is returned, the JWT is verified
+    1. If successful, the request is treated as signed in
+8. If an updated session JWT is not returned, the request is treated as signed out
+
+## Scenarios that trigger a handshake
+
+<Callout type="warning">
+A handshake is only triggered If a request is determined to be a document request (detected by the presence of the `Sec-Fetch-Dest: document` header, or `Accept: text/html`). Otherwise, the request is treated as `signed-out`.
+</Callout>
+
+1. Development instance and a dev browser is detected in URL **[URL-based session sync]**
+2. Production instance and request is being made to a satellite application **[satellite needs sync]**
+3. Has a session token (`__session`), but no active client is detected (`__client_uat` = 0 or missing) **[session token without client UAT]**
+4. Has active client (`__client_uat` > 0) but no session token **[client UAT without session token]**
+5. Client UAT is more recent than current session token issued at (`__client_uat` > `sessionToken.iat`) **[session token outdated]**
+6. Session token exists but is expired **[session token expired]**
+7. Session token exists but is not active yet **[session token not active yet]**
+
+## Handshake architecture
+
+The handshake flow is composed of two main pieces:
+
+1. The FAPI [`/client/handshake` endpoint](https://github.com/clerk/clerk_go/blob/main/api/fapi/v1/router/router.go#L223)
```

Comment:
Please replace this link that points to our internal Go repo with a link to the [FAPI API Reference](https://clerk.com/docs/reference/frontend-api/). 

The FAPI API reference doesn't contain the handshake endpoint yet but @Nikpolik can help us with that.

Reply:
Sorry for this! PR to fix this https://github.com/clerk/clerk_go/pull/5243